azurerm/internal/services/managedapplications/managed_application_resource.go

```diff
@@ -0,0 +1,313 @@
+	}
+
+	if v, ok := d.GetOk("managed_resource_group_name"); ok {
+		targetResourceGroupId := fmt.Sprintf("/subscriptions/%s/resourceGroups/%s", meta.(*clients.Client).Account.SubscriptionId, v)
+		parameters.ApplicationProperties = &managedapplications.ApplicationProperties{
+			ManagedResourceGroupID: utils.String(targetResourceGroupId),
+		}
+	}
+	if v, ok := d.GetOk("application_definition_id"); ok {
+		a := v.(string)
+		b := utils.String(a)
+		parameters.ApplicationDefinitionID = b
+	}
+	if v, ok := d.GetOk("plan"); ok {
+		parameters.Plan = expandManagedApplicationPlan(v.([]interface{}))
+	}
+	if v, ok := d.GetOk("parameters"); ok {
+		params := v.(map[string]interface{})
+
+		newParams := make(map[string]interface{}, len(params))
+		for key, val := range params {
+			newParams[key] = struct {
+				Value interface{} `json:"value"`
+			}{
+				Value: val,
+			}
+		}
+
+		parameters.Parameters = &newParams
+	}
+
+	future, err := client.CreateOrUpdate(ctx, resourceGroupName, name, parameters)
+	if err != nil {
+		return fmt.Errorf("failed to create Managed Application %q (Resource Group %q): %+v", name, resourceGroupName, err)
+	}
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("failed to wait for creation of Managed Application %q (Resource Group %q): %+v", name, resourceGroupName, err)
+	}
+
+	resp, err := client.Get(ctx, resourceGroupName, name)
+	if err != nil {
+		return fmt.Errorf("failed to retrieve Managed Application %q (Resource Group %q): %+v", name, resourceGroupName, err)
+	}
+	if resp.ID == nil || *resp.ID == "" {
+		return fmt.Errorf("cannot read Managed Application %q (Resource Group %q) ID", name, resourceGroupName)
+	}
+	d.SetId(*resp.ID)
+
+	return resourceManagedApplicationRead(d, meta)
+}
+
+func resourceManagedApplicationRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).ManagedApplication.ApplicationClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.ManagedApplicationID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Get(ctx, id.ResourceGroup, id.Name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] Managed Application %q does not exist - removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("failed to read Managed Application %q (Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
+	}
+
+	d.Set("name", id.Name)
+	d.Set("resource_group_name", id.ResourceGroup)
+	d.Set("location", location.NormalizeNilable(resp.Location))
+	d.Set("kind", resp.Kind)
+	if err := d.Set("plan", flattenManagedApplicationPlan(resp.Plan)); err != nil {
+		return fmt.Errorf("setting `plan`: %+v", err)
+	}
+	if props := resp.ApplicationProperties; props != nil {
+		parsedManagedResourceGroupID := strings.Split(*props.ManagedResourceGroupID, "/")
```

Comment:
could we use the resource group parse id function here maybe?

Reply:
updated